test/distributed/org/apache/cassandra/distributed/impl/CollectingSslRMIServerSocketFactoryImpl.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.distributed.impl;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.net.SocketException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocket;
+import javax.net.ssl.SSLSocketFactory;
+
+import org.apache.cassandra.utils.RMICloseableServerSocketFactory;
+
+
+/**
+ * This class is used to keep track of SSL based RMI servers created during a cluster creation to
+ * later close the sockets, which would otherwise be left with a thread running waiting for
+ * connections that would never show up as the server was otherwise closed.
+ */
+class CollectingSslRMIServerSocketFactoryImpl implements RMICloseableServerSocketFactory
+{
+    private final InetAddress bindAddress;
+    private final String[] enabledCipherSuites;
+    private final String[] enabledProtocols;
+    private final boolean needClientAuth;
+    private SSLContext sslContext;
```

Comment:
If we change this field to be:
```suggestion
    private final SSLSocketFactory sslSocketFactory;
```
We can then initialize it using the `sslContext` passed in to the constructor that uses it, _or_ use `getDefaultSSLSocketFactory` in the other, which gets rid of the ternary operator in `createSslServerSocket`. Do you envision a need to hold on to the `sslContext` itself, or could we do this in stead?

Reply:
Ack. Will pick it up.